tutorials/xr/openxr_spatial_entities.rst

```diff
@@ -0,0 +1,1710 @@
+                    new_scene.free()
+        elif tracker is OpenXRPlaneTracker:
+            if plane_tracker_scene:
+                var new_scene = plane_tracker_scene.instantiate()
+                if new_scene is XRAnchor3D:
+                    new_node = new_scene
+                else:
+                    push_error("Plane tracking scene doesn't have an XRAnchor3D as a root node and can't be used!")
+                    new_scene.free()
+        elif tracker is OpenXRMarkerTracker:
+            if marker_tracker_scene:
+                var new_scene = marker_tracker_scene.instantiate()
+                if new_scene is XRAnchor3D:
+                    new_node = new_scene
+                else:
+                    push_error("Marker tracking scene doesn't have an XRAnchor3D as a root node and can't be used!")
+                    new_scene.free()
+        else:
+            # Type of spatial entity tracker we're not supporting?
+            push_warning("OpenXR Spatial Entities: Unsupported anchor tracker " + tracker.get_name() + " of type " + tracker.get_class())
+
+        if not new_node:
+            # No scene defined or able to be instantiated? We're done!
+            return
+
+        # Setup and add to our scene.
+        new_node.tracker = tracker.name
+        new_node.pose = "default"
+        _managed_nodes[tracker] = new_node
+        add_child(new_node)
+
+        added_spatial_entity.emit(new_node)
+
+
+    # A new tracker was added to our XRServer.
+    func _on_tracker_added(tracker_name: StringName, type: int):
+        if type == XRServer.TRACKER_ANCHOR:
+            var tracker : XRTracker = XRServer.get_tracker(tracker_name)
+            if tracker and tracker is OpenXRSpatialEntityTracker:
+                _add_tracker(tracker)
+
+
+    # A tracked managed by XRServer was changed.
+    func _on_tracker_updated(_tracker_name: StringName, _type: int):
+        # For now we ignore this, there aren't changes here we need to react
+        # to and the instanced scene can react to this itself if needed.
+        pass
+
+
+    # A tracker was removed from our XRServer.
+    func _on_tracker_removed(tracker_name: StringName, type: int):
+        if type == XRServer.TRACKER_ANCHOR:
+            var tracker : XRTracker = XRServer.get_tracker(tracker_name)
+            if _managed_nodes.has(tracker):
+                # We emit this right before we remove it!
+                removed_spatial_entity.emit(_managed_nodes[tracker])
+
+                # Remove the node.
+                remove_child(_managed_nodes[tracker])
+
+                # Queue free the node.
+                _managed_nodes[tracker].queue_free()
+
+                # And remove from our managed nodes.
+                _managed_nodes.erase(tracker)
+
+Spatial anchors
+---------------
+
+Spatial anchors allow us to map real world location in our virtual world in such a way that the
+XR runtime will keep track of these locations and adjust them as needed.
```

Comment:
Use of singular "location" in "real world location" but use of plural "locations" in "... will keep track of these locations"